packages/adapter-prisma/src/adapter-prisma.js

```diff
@@ -26,22 +22,27 @@ class PrismaAdapter {
 
   async connect({ rels }) {
     // Connect to the database
-    await this._connect({ rels }, this.config);
+    // the adapter was already connected since we have a prisma client
+    // it may have been disconnected since it was connected though
+    // so connect but don't regenerate the prisma client
+    if (this.prisma) {
+      await this.prisma.$connect();
+      return;
+    }
+    if (!this.config.prismaClient) {
+      throw new Error('You must pass the prismaClient option to connect to a database');
+    }
+    this.prisma = new this.config.prismaClient({
+      log: this.enableLogging && ['query'],
+      datasources: { [this.provider]: { url: this.url } },
+    });
+    await this.prisma.$connect();
 
     // Set up all list adapters
     try {
-      // Validate the minimum database version requirements are met.
-      await this.checkDatabaseVersion();
-
-      const taskResults = await this.postConnect({ rels });
-      const errors = taskResults.filter(({ isRejected }) => isRejected).map(({ reason }) => reason);
-
-      if (errors.length) {
-        if (errors.length === 1) throw errors[0];
-        const error = new Error('Multiple errors in PrismaAdapter.postConnect():');
-        error.errors = errors;
-        throw error;
-      }
+      Object.values(this.listAdapters).forEach(listAdapter => {
+        listAdapter._postConnect({ rels, prisma: this.prisma });
+      });
```

Comment:
Could you move this to before the connect(and you might want to rename _postConnect too then)? Since then you can remove the catching and disconnecting stuff below and it's important for the Next-lite stuff that the list APIs can be called synchronously(even though they'll ofc return asynchronously) and moving it to before the connect enables that (in that PR currently, I call postConnect on the adapter to make it work)

Reply:
Cool, will do 👍